### ISSUE TITLE
CLI doesn't display ALWAYS errors durning build/serve process

### ISSUE THREAD

**Issue by istiti** [angular/angular-cli#5931]
### Bug Report or Feature Request (mark with an `x`)
```
- [x ] bug report -> please search issues before submitting
- [ ] feature request
```

### Versions.
```
@angular/cli: 1.0.0
node: 7.7.2
os: win32 x64
@angular/common: 4.0.1
@angular/compiler: 4.0.1
@angular/core: 4.0.1
@angular/forms: 4.0.1
@angular/http: 4.0.1
@angular/platform-browser: 4.0.1
@angular/platform-browser-dynamic: 4.0.1
@angular/router: 4.0.1
@angular/cli: 1.0.0
@angular/compiler-cli: 4.0.1
```

I want to import correctly my rxjs operator and instead importing whole "rxjs" lib I do `import {Observable} from 'rxjs/Observable';` the problem is sometimes cli temm me in the console that this or this operator doesn't exist and break the build process but somes times cli doesn't tell me errors even if I don't import it explicitly. 
 So my follow code break only when running on browser after "ng b --prod" 
Here iy my imports: 
```
import {DataEventModification, DataEventType, SharedService} from '../shared.service';
import {ElementRef, EventEmitter, HostListener, OnDestroy, OnInit, Output} from '@angular/core';
import {TSABEntityType} from './lockable-form/form-locker.service';
import {FormGroup} from '@angular/forms';
import {Observable} from 'rxjs/Observable';
import {Subscription} from 'rxjs/Subscription';
import {DialogService} from '../dialog/dialog.service';
```

my code is something like : 
```
if( this.formFilters) {
      this.formFiltersSubscriber = this.formFilters.valueChanges
        .debounceTime(this.filterDebounceTime)
        .skip(1)
        .subscribe(data => {
          this.filters = Object.assign(this.filters, data);

          for(let index in this.filters)
            if( this.filters[index] == "" || this.filters[index] == null)
              delete this.filters[index];

          this.saveTableState();

          this.reloadServerData(this.filters);
        });

      this.formFilters.reset();
      this.formFilters.patchValue(this.filters);
    }
```

skip is available an debounceTime too for webstorm and my angular-cli sucess compiling well using AoT!!  without errors BUT when I run app on browser I get this error : `TypeError: this.formFilters.valueChanges.debounceTime(...).skip is not a function`

All I need is import manually these two things : 
`import "rxjs/add/operator/skip"; import "rxjs/add/operator/debounceTime";`

The problem is not about import manually the problem is it only break on browser with AoT, this is strange, and I need build using --prod flag it take a while then run this build on browser and I can get this error only at this moment. 

 
I admit with this kind of code : 
```
return dialogRef.afterClosed().flatMap(res => {
        switch (res ) {
          case 'cancel':
            return Observable.of(false);
          case 'continue':

            if( !comp.isDataValid())
              return Observable.of(false);

            return comp.getFormLocker().applyModifications(false, true).map(() => {
              return true;
            }).catch(()=> {
              return Observable.of(false)
            });

          case 'unlockAndContinue':
            comp.getFormLocker().abortModifications(true);
            return Observable.of(true);
        }
      }).first();
```
when I compile ng serve or ng build, cli tell me this great error `ERROR in C:/Users/.../src/app/can-deactivate-entity.guard.ts (51,10): Property 'first' does not exist on type 'Observable<{}>'.` but not on code above

Is this bug ?

**Comment by hansl** [comment]
RxJS imports have 2 side effects:

1. add the types to the global type system. This means you'll see it in WebStorm and it will not error when doing type checking, and
2. add the function _at the time of import_ to the prototype.

The problem you see is there is a file that import the skip operator, but that file is not loaded by your browser.

This is a global problem with TypeScript global type declarations and something recurring with the pattern RxJS uses to declare operators. We suggest you put all your operators in `polyfills.ts` to fix it, or you _always_ import it everywhere you actually use it.

Long story short, importing an `rxjs/add/operator/...` _anywhere_ will add the operator to the RxJS types, but it might not be available in the files you're using it.

**Comment by istiti** [comment]
I did what you said (all operators in `polyfills.ts`) and I don't understand why rxjs is imported in pollyfill and vendor bundle, any hits ? 
## see screenshot

![rxjs2times](https://cloud.githubusercontent.com/assets/2789203/25167641/8483fc76-24e0-11e7-8b8c-ba83a7e0db67.JPG)
